components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/DefaultRequestCoordinator.java

```diff
@@ -393,6 +396,23 @@ && isStepHasMultiOption(context)
 
                 if (!context.isLogoutRequest()) {
                     FrameworkUtils.getAuthenticationRequestHandler().handle(request, responseWrapper, context);
+                    if (!ORGANIZATION_AUTHENTICATOR.equals(request.getParameter(AUTHENTICATOR))) {
+                        String redirectURL = responseWrapper.getRedirectURL();
+                        URI uri = new URI(redirectURL);
+                        String query = uri.getRawQuery();
+                        if (StringUtils.isNotBlank(query)) {
+                            if (!query.contains(FrameworkConstants.REQUEST_PARAM_SP_UUID + "=")) {
+                                redirectURL = redirectURL + "&" + FrameworkConstants.REQUEST_PARAM_SP_UUID
+                                        + "=" + URLEncoder.encode(context.getServiceProviderResourceId(),
+                                        StandardCharsets.UTF_8.name());
+                            }
+                        } else {
+                            redirectURL = redirectURL + "?" + FrameworkConstants.REQUEST_PARAM_SP_UUID
+                                    + "=" + URLEncoder.encode(context.getServiceProviderResourceId(),
+                                    StandardCharsets.UTF_8.name());
+                        }
+                        responseWrapper.sendRedirect(redirectURL);
```

Comment:
is it correct to do a redirection here?

Reply:
it's not redirecting. It's only setting the redirect url in the responseWrapper object. The function name is mileading.